hgtv/views/playlist.py

```diff
@@ -230,45 +248,36 @@ def playlist_import(channel):
-            flash(e.message, category="error")
-            return render_form(form=form, title="Import Playlist", submit=u"Import",
-                cancel_url=channel.url_for(), ajax=False)
-        flash(u"Imported playlist '%s'." % playlist.title, 'success')
-        return render_redirect(playlist.url_for(), code=303)
-    return render_form(form=form, title="Import Playlist", submit=u"Import",
-        cancel_url=channel.url_for(), ajax=False)
+            return make_response(jsonify(status='error', errors={'error': [e.message]}), 400)
+    return make_response(jsonify(status='error', errors=form.errors), 400)
 
 
 @app.route('/<channel>/<playlist>/extend', methods=['GET', 'POST'])
 @lastuser.requires_login
+@render_with({'text/html': 'index.html.jinja2'}, json=True)
 @load_models(
     (Channel, {'name': 'channel'}, 'channel'),
     (Playlist, {'name': 'playlist', 'channel': 'channel'}, 'playlist'),
     permission='extend')
 def playlist_extend(channel, playlist):
     form = PlaylistImportForm()
     form.channel = channel
-    html = render_template('playlist-extend.html.jinja2', form=form, channel=channel, playlist=playlist)
-    if request.is_xhr:
-        if form.validate_on_submit():
-            playlist_url = escape(form.playlist_url.data)
-            initial_count = len(playlist.videos)
-            try:
-                process_playlist(playlist_url=playlist_url, playlist=playlist)
-            except:
-                return jsonify({'message_type': "server-error",
-                    'message': 'Oops, something went wrong, please try later'})
-            additions = (len(playlist.videos) - initial_count)
-            if additions:
-                db.session.commit()
-                cache.delete('data/featured-channels')
-                flash(u"Added '%d' videos" % (len(playlist.videos) - initial_count), 'success')
-                return jsonify({'message_type': "success", 'action': 'redirect', 'url': playlist.url_for()})
-            return jsonify({'message_type': "success", 'action': 'noop', 'message': 'Already upto date'})
-        if form.errors:
-            html = render_template('playlist-extend.html.jinja2', form=form, channel=channel, playlist=playlist)
-            return jsonify({'message_type': "error", 'action': 'append',
-                'html': html})
-        return jsonify({'action': 'modal-window', 'message_type': 'success', 'html': html})
-    return html
+    if request.method == 'GET':
+        html_form = render_form(form=form, title=u"Playlist extend", submit=u"Save",
+        cancel_url=playlist.url_for(), ajax=False, with_chrome=False)
+        return {'playlist': dict(playlist.current_access()), 'form': html_form}
+    if form.validate_on_submit():
+        playlist_url = escape(form.playlist_url.data)
+        initial_count = len(playlist.videos)
+        try:
+            process_playlist(playlist_url=playlist_url, playlist=playlist)
+        except Exception as e:
```

Comment:
Replace `Exception` with something more specific. If you're not sure which exception, remove it entirely. We need error logs we can analyse.